tests/unit/lms/views/dashboard/pagination_test.py

```diff
@@ -0,0 +1,87 @@
+import json
+
+import pytest
+from h_matchers import Any
+from sqlalchemy import select
+
+from lms.js_config_types import Pagination
+from lms.models import Course
+from lms.validation import ValidationError
+from lms.views.dashboard.pagination import PaginationParametersMixin, get_page
+from tests import factories
+
+
+class TestGetPage:
+    def test_when_no_next_page(self, pyramid_request, db_session):
+        pyramid_request.parsed_params = {"limit": 100}
+        courses = factories.Course.create_batch(5)
+        query = select(Course)
+        db_session.flush()
+
+        items, pagination = get_page(pyramid_request, query, (Course.id,))
+
+        assert items == courses
+        assert pagination == Pagination(next=None)
+
+    def test_when_empty(self, pyramid_request):
+        pyramid_request.parsed_params = {"limit": 100}
+        query = select(Course).where(False)
+
+        items, pagination = get_page(pyramid_request, query, (Course.id,))
+
+        assert items == []
+        assert pagination == Pagination(next=None)
+
+    def test_calculates_next(self, pyramid_request, db_session):
+        pyramid_request.parsed_params = {"limit": 1}
+        courses = factories.Course.create_batch(5)
+        query = select(Course)
+        db_session.flush()
+
+        items, pagination = get_page(pyramid_request, query, (Course.id,))
+
+        assert items == courses[0:1]
+        assert pagination == Pagination(
+            next=Any.url.with_query({"cursor": json.dumps([courses[0].id])})
+        )
+
+    def test_filters_by_cursor(self, pyramid_request, db_session):
+        courses = factories.Course.create_batch(5)
+        query = select(Course)
+        db_session.flush()
+        pyramid_request.parsed_params = {
+            "cursor": [courses[0].id, courses[0].lms_name],
+            "limit": 1,
+        }
+
+        items, _ = get_page(pyramid_request, query, (Course.id, Course.lms_name))
+
+        assert items == courses[1:2]
+
+
+class TestPaginationParametersMixin:
+    def test_limit_default(self, pyramid_request):
+        assert PaginationParametersMixin(pyramid_request).parse() == {"limit": 100}
+
+    @pytest.mark.parametrize(
+        "cursor",
+        [
+            pytest.param("1,", id="Not JSON"),
```

Comment:
```suggestion
            pytest.param("1,", id="Not a list"),
```

Strings containing JSON primitives (eg. "1", "true", "NaN") _are_ valid JSON, they just deserialize to a different kind of value than the code expects.